nservicebus/recoverability/index.md

```diff
@@ -39,6 +39,8 @@ By default up to five immediate retries are performed if the message processing
 
 Note: The configured value describes the minimum number of times a message will be retried if its processing consistently fails. Especially in environments with competing consumers on the same queue, there is an increased chance of retrying a failing message more times across the endpoints.
 
+Note: Depending on the concurrency and transactionallity settings of the endpoint immediate retries might happen even if configured to be off. To ensure that no retries is performed the endpoints needs to be configured to be either [non transactional](/nservicebus/transports/transactions.md) or the [concurrency retricted to 1](/nservicebus/operations/tuning.md).
```

Comment:
I thik that caveats are more caused by transport implementations than the transactionality. Here is my take on them:
 * noTx: 
   * 0 retries - caveat is we can have 0 processing attempts
 * receive counter stored in memory cache (MSMQ, SQL) and retires configured to x:
   * inf. retries - the cache for counter is invalidated or endpoint re-starts
   * more than x retries but finite 
      * transaction timeouts with too long running handlers
      * scaled-out instances and counter duplicates (this applies only to SQL)
  * receive counter manged by the broker (ASB, ASP) and retries configured to x:
      *  anything between 0 and x - endpoint re-start with non empty prefetch buffers
  * in-memory/per-message (RabbitMQ)
      * anything from 0 to inf. - endpoint re-starts